src/js/Content/Features/Store/Common/AddToCart.js

```diff
@@ -1,61 +1,45 @@
-import {Localization, SyncedStorage} from "../../../../modulesCore";
-import {ConfirmDialog, RequestData} from "../../../modulesContent";
+import {Localization} from "../../../../modulesCore";
+import {RequestData} from "../../../modulesContent";
 import {Page} from "../../Page";
 
 export class AddToCart {
 
     static async post(formEl, onWishlist, addToCartEl) {
 
-        const cartUrl = formEl.getAttribute("action");
-        const addToCartStr = Localization.str.addtocart_dialog;
+        // Use the form's `action` here because free promotions are submitted to a different endpoint
+        const endpoint = formEl.getAttribute("action");
+        const body = new FormData(formEl);
+
         let response;
 
         try {
-            response = await RequestData.post(cartUrl, new FormData(formEl), {}, "none");
+            response = await RequestData.post(endpoint, body, {}, "none");
         } catch (err) {
             // Likely network error; response.ok doesn't mean the item was added successfully
             Page.runInPageContext((errTitle, errDesc, errStr) => {
                 window.SteamFacade.showAlertDialog(errTitle, `${errDesc}<br><br>${errStr}`);
             },
             [
                 Localization.str.error,
-                addToCartStr.error_desc,
+                Localization.str.addtocart_dialog.error_desc,
                 err.toString()
             ]);
 
             return;
         }
 
-        // If redirected to a page other than the cart, follow the redirect
-        if (response.url !== cartUrl) {
+        /**
+         * If redirected to a page other than the cart, follow the redirect.
+         * E.g. CSGO operations redirect to a /approvetxn/ page.
+         */
+        const url = new URL(response.url);
+        if (!/^\/cart\/?$/.test(url.pathname)) {
             window.location.assign(response.url);
             return;
         }
 
-        let enabled = SyncedStorage.get("addtocart_no_redirect");
-
-        // Show feature hint to first time users
-        if (!SyncedStorage.has("addtocart_no_redirect")) {
-
-            enabled = await ConfirmDialog.openFeatureHint(
-                "addtocart_no_redirect",
-                addToCartStr.title,
-                addToCartStr.desc,
-                addToCartStr.continue,
-                addToCartStr.checkout
-            ) === "OK";
-
-            SyncedStorage.set("addtocart_no_redirect", enabled);
-        }
```

Comment:
This removes the feature hint from DLC checkboxes, doesn't it?

Reply:
Should be fixed now.